src/OpenDiffix.Core/Analyzer.fs

```diff
@@ -419,13 +424,30 @@ let rec private normalizeBucketLabelExpression expression =
     FunctionExpr(ScalarFunction fn, List.map normalizeBucketLabelExpression args @ extraArgs)
   | _ -> expression
 
+let private untrustedAllowsRange arg =
+  match arg with
+  // "money-style" numbers, i.e. 1, 2, or 5 preceeded by or followed by zeros: ⟨... 0.1, 0.2, 0.5, 1, 2, 5, 10, 20, ...⟩
+  | Constant (Real c) -> Regex("^(1|2|5)\.0+e(\-|\+)[0-9]+$").Match(sprintf "%.15e" c).Success
```

Comment:
I considered a more "mathy" formulation of this condition, but deciding this through a string representation seemed more promising in terms of clarity.

Reply:
Interesting! I'm not sure how to judge this, I guess `*.5` plays well with binary representation.

Some notes about regex use:

- Pattern should be cached outside of the function, or use `RegExp.IsMatch(string, pattern)`.
- Use `IsMatch` instead of `Match().Success`
- Groups don't need to capture since you only check for match success
- `(1|2|5)` can be simplified to `[125]`
- `[0-9]` can be simplified to `\d`
- `(\-|\+)` can be simplified to `[-+]`

Reply:
> I guess *.5 plays well with binary representation.

yeah, the others are a bit funky, hence the `%.15e`. I'm not 100% it's completely air tight, but should be good enough, even if some edge cases :thinking:.

The alternative would be to disassemble the floating point binary representation and dig into that. The regex version has the advantage that it reads "similar" to the definition from the specs - "if it looks like money-style number, it is a money-style number"

> Some notes about regex use

Thx!

Reply:
Do you intend to push any change for this?

Reply:
missed this one, crazy day today :bow: And I think gh sometimes shows it for me and sometimes doesn't...